testSrc/com/intellij/plugins/haxe/ide/HaxeSemanticAnnotatorTest.java

```diff
@@ -178,4 +178,13 @@ public void testImplementSignature() throws Exception {
   public void testSimpleAssignUnknownGeneric() throws Exception {
     doTestNoFixWithWarnings();
   }
+
+  public void testExtendsAnonymousType() throws Exception {
+    doTestNoFixWithWarnings();
+  }
+
+  // TODO: test never completes (that cases works in plugin, but just not work in unit-tests environment)
+  //public void testExtendsSelf() throws Exception {
+  //  doTestNoFixWithWarnings();
+  //}
```

Comment:
See https://github.com/eliasku/intellij-haxe/pull/3
Caused by probably some recursion in HaxeClass.getExtendsList and similar in AbstractTypedef
Eliminated by excluding self from extends